graphics/mupdf/Portfile

```diff
@@ -34,21 +34,24 @@ homepage            https://www.mupdf.com/
 master_sites        ${homepage}downloads/archive/
 distname            ${name}-${version}-source
 
-checksums           rmd160  53015efda97139e3e3308fd1ed86ade1631d9fbc \
-                    sha256  54c66af4e6ef8cea9867cc0320ef925d561b42919ea0d4f89db5c9ef485bbeb7 \
-                    size    57541621
+checksums           rmd160  d3c2420343f8f0ce051134133e94bd5a2f2f4f64 \
+                    sha256  1ef9a6409bc0a3271586e1b16f78eb156a579521cd212a124b57c4da8b940aad \
+                    size    55144800
 
 depends_build       port:pkgconfig
 
 depends_lib         port:freetype \
                     port:gumbo-parser \
                     path:lib/pkgconfig/harfbuzz.pc:harfbuzz \
+                    port:leptonica \
                     port:jbig2dec \
                     path:include/turbojpeg.h:libjpeg-turbo \
                     port:openjpeg \
+                    port:openssl3 \
```

Comment:
mupdf uses libcrypto for PDF signature support. Its build detects libcrypto by using `pkg-config`, so if this port was built when OpenSSL was already installed, it would have a dependency on OpenSSL, even though the Portfile didn’t declare it. This addition fixes that problem by declaring the dependency.

I switched it to use `PortGroup` at your request.

Reply:
thanks for that explanation; avoiding opportunistic linking is always a good idea, so adding that dependency then makes sense!